dask_ml/model_selection/_successive_halving.py

```diff
@@ -0,0 +1,215 @@
+
+        The values in the ``test_score`` key correspond to the last score a model
+        received on the hold out dataset. The key ``model_id`` corresponds with
+        ``history_``. This dictionary can be imported into Pandas.
+
+    model_history_ : dict of lists of dict
+        A dictionary of each models history. This is a reorganization of
+        ``history_``: the same information is present but organized per model.
+
+        This data has the structure  ``{model_id: hist}`` where ``hist`` is a
+        subset of ``history_`` and ``model_id`` are model identifiers.
+
+    history_ : list of dicts
+        Information about each model after each ``partial_fit`` call. Each dict
+        the keys
+
+        * ``partial_fit_time``
+        * ``score_time``
+        * ``score``
+        * ``model_id``
+        * ``params``
+        * ``partial_fit_calls``
+
+        The key ``model_id`` corresponds to the ``model_id`` in ``cv_results_``.
+        This list of dicts can be imported into Pandas.
+
+    best_estimator_ : BaseEstimator
+        The model with the highest validation score among all the models
+        retained by the "inverse decay" algorithm.
+
+    best_score_ : float
+        Score achieved by ``best_estimator_`` on the vaidation set after the
+        final call to ``partial_fit``.
+
+    best_index_ : int
+        Index indicating which estimator in ``cv_results_`` corresponds to
+        the highest score.
+
+    best_params_ : dict
+        Dictionary of best parameters found on the hold-out data.
+
+    scorer_ :
+        The function used to score models, which has a call signature of
+        ``scorer_(estimator, X, y)``.
+
+    n_splits_ : int
+        Number of cross validation splits.
+
+    multimetric_ : bool
+        Whether this cross validation search uses multiple metrics.
+
+
+    References
+    ----------
+    .. [1] "Non-stochastic best arm identification and hyperparameter
+           optimization" by Jamieson, Kevin and Talwalkar, Ameet. 2016.
+           https://arxiv.org/abs/1502.07943
+
+    """
+
+    def __init__(
+        self,
+        estimator,
+        parameters,
+        n_initial_parameters=10,
+        n_initial_iter=9,
+        aggressiveness=3,
+        max_iter=100,
+        test_size=None,
+        patience=False,
+        tol=1e-3,
+        random_state=None,
+        scoring=None,
+    ):
+        self.n_initial_parameters = n_initial_parameters
+        self.n_initial_iter = n_initial_iter
+        self.aggressiveness = aggressiveness
+
+        super(SuccessiveHalvingSearchCV, self).__init__(
+            estimator,
+            parameters,
+            n_initial_parameters=n_initial_parameters,
+            max_iter=max_iter,
+            test_size=test_size,
+            patience=patience,
+            tol=tol,
+            random_state=random_state,
+            scoring=scoring,
+        )
+
+    def _adapt(self, info, _first_step_taken=False):
```

Comment:
No need for `_first_step_taken` to be prefixed w/ an underscore. Maybe just call it `is_first_step`.